vnext/Microsoft.ReactNative.Managed/ReflectionReactPackageProvider.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.ReactNative;
+
+namespace Microsoft.ReactNative.Managed
+{
+  /// <summary>
+  /// A default implementation that uses reflection to extract the Modules and ViewManagers
+  /// to be registered on the package builder.
+  /// </summary>
+  /// <typeparam name="T">This must be thea type defined in the assembly where the items for the react package should be extracted.</typeparam>
+  public class ReflectionReactPackageProvider<T> : IReactPackageProvider
+  {
+    public virtual void CreatePackage(IReactPackageBuilder packageBuilder)
+    {
+      ReflectionReactPackageProviderExtensions.AddReflectionReactPackageProvider<T>(packageBuilder);
```

Comment:
>AddReflectionReactPackageProvider [](start = 47, length = 33)

Just curious: why not to call directly on the packageBuilder?

Reply:
Reuse of code. Sometimes in the 'App's you want to register this class. `Libraries' need to implement their own `IReactPackageProvider` and it needs to be a WinRT type, so it can't just inherit from `ReflectionReactPackageProvider<T>` because it breaks winRT. 

I initially had their implementation call `new RelfectionReactPacakgeProvider<PackageProvider>().CreatePackage(packageBuilder)` but then opted for the extension method to be a little bit cleaner.

I'm still not a fan of having both but with the current model this is the cleanest I could get it. When we start doing codgen for managed code I beleive we can clean things up a bit more.


---
In reply to: [420384668](https://github.com/microsoft/react-native-windows/pull/4793#discussion_r420384668) [](ancestors = 420384668)

Reply:
My question was why we call there the extension method as a static method, and not using it as an extension method. Changing this line to
``` C#
packageBuilder.AddReflectionReactPackageProvider<T>();
```
must not have any difference besides making it more compact. Am I missing something?